lib/rules/selector-max-specificity/index.cjs

```diff
@@ -27,29 +29,127 @@ const meta = {
- *
- * @returns {Specificity}
+ * @param {import('postcss-selector-parser').Selector} selector
+ * @param {(selector: string) => boolean} isIgnored
+ * @returns {import('postcss-selector-parser').Selector}
  */
-const zeroSpecificity = () => ({ a: 0, b: 0, c: 0 });
+function removeIgnoredNodes(selector, isIgnored) {
+	const clone = selector.clone();
+
+	clone.walk((node) => {
+		let value = '';
+
+		switch (node.type) {
+			case 'attribute':
+			case 'class':
+			case 'id':
+			case 'tag':
+				value = node.toString();
+				break;
+			case 'pseudo':
+				value = node.value;
+				break;
+
+			default:
+				return
+		}
+
+		if (!value) {
+			return;
+		}
+
+		if (isIgnored(value)) {
+			node.replaceWith(selectorParser.universal({
+				value: '*',
+				sourceIndex: node.sourceIndex,
+				source: node.source,
+			}));
+		}
+	});
+
+	return clone;
+}
 
 /**
- * Calculate the sum of given specificities.
- *
- * @param {Specificity[]} specificities
- * @returns {Specificity}
+ * @typedef {import('postcss-selector-parser').Selector} Selector
+ * @typedef {import('postcss-selector-parser').Root} SelectorRoot
+ * @param {import('postcss').Rule} rule
+ * @param {import('stylelint').PostcssResult} result
+ * @param {(selector: string) => boolean} isIgnored
+ * @returns {Array<{selector: Selector, resolvedSelectors: Array<Selector>}> | undefined}
  */
-const specificitySum = (specificities) => {
-	const sum = zeroSpecificity();
+function gatherSelectorsForRule(rule, result, isIgnored) {
+	/** @typedef {import('postcss').Document} Document */
+	/** @typedef {import('postcss').Root} Root */
+	/** @typedef {import('postcss').Container} Container */
+
+	const ownAST = selectorAST(getRuleSelector(rule), result, rule);
+
+	if (!ownAST) return;
 
-	for (const { a, b, c } of specificities) {
-		sum.a += a;
-		sum.b += b;
-		sum.c += c;
+	/** @type {import('postcss-selector-parser').Root | undefined} */
+	let ast = undefined;
+
+	/** @type {Array<Document|Root|Container>} */
+	let ancestors = [];
+
+	{
+		/** @type {Document|Root|Container|undefined} */
+		let parent = rule.parent;
+
+		while (parent) {
+			ancestors.push(parent);
+			parent = parent.parent;
+		}
 	}
 
-	return sum;
-};
+	ancestors.reverse();
+
+	for (const child of ancestors) {
+		if (typeGuards.isRule(child)) {
+			const childAST = selectorAST(getRuleSelector(child), result, child);
+
+			if (!childAST) return;
+
+			if (ast) {
+				ast = selectorResolveNested.resolveNestedSelector(childAST, ast);
+			} else {
+				ast = childAST;
+			}
+
+			continue;
+		}
+
+		if (typeGuards.isAtRule(child) && ast) {
+			// `.foo, #bar { @media screen { color: red; } }`
+			// equivalent to
+			// `@media screen { .foo, #bar { & { color: red; } } }`
+			// `@media screen { :is(.foo, #bar) { color: red; } }`
+			const childAST = selectorAST('&', result, child);
+
+			if (!childAST) return;
+
+			ast = selectorResolveNested.resolveNestedSelector(childAST, ast);
+		}
```

Comment:
for more info and demo's see : https://github.com/mrhenry/stylelint-mrhenry/issues/66